src/v/cluster/tm_stm.h

```diff
@@ -72,6 +74,7 @@ struct tm_transaction {
     model::term_id etag;
     tx_status status;
     std::chrono::milliseconds timeout_ms;
+    model::timestamp last_update_ts;
```

Comment:
you might consider using `seastar::lowres_system_clock` here. it will still give you wall clock time, but it is cheap to query compared to model::timestamp's std::chrono::system_clock.

on a related note: you might consider writing a few comments describing the requirement of system clock time which doesn't feel obvious to me. in particular, it seems that (1) transactions with a timestamp stored to a topic may be read on a different node where they are GCd thus requiring some global time representation and that (2) this implies that we should make sure to note that NTP clock synchronization is important in a redpanda cluster?

Reply:
good idea, I tried to use `ss::lowres_clock` but it has opaque epoch which doesn't match system epoch so had to ditch it, but `ss::lowres_system_clock` doesn't have this problem, switchig to it

Reply:
also updated the commit message about the cluster-wide clock skew, TLDR: skewed clocks don't cause safety violation, default expiration time is a week so it's fine if the clocks drift up to several hours